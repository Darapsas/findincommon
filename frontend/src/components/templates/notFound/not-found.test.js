import React from 'react'
import { shallow, mount } from 'enzyme'
import NotFound from './not-found'
import { MemoryRouter, Link } from 'react-router-dom'
import renderer from 'react-test-renderer'

describe('<NotFound />', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NotFound />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('button exists', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/not-found']}>
        <NotFound />
      </MemoryRouter>
    )
    const linkButton = wrapper.find('button')
    expect(linkButton).toBeDefined()
  })
})
