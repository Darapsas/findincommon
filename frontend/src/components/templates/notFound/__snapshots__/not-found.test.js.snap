// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> matches the snapshot 1`] = `
<div
  className="page-not-found"
>
  <h1
    className="title"
  >
    404
  </h1>
  <div
    className="desc"
  >
    The Page you're looking for was not found.
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="go-back-btn btn btn-primary"
      type="button"
    >
      Go Back
    </button>
  </a>
</div>
`;
